### PR TITLE
std.numeric: Support mixing integral types in gcd()

### PR DESCRIPTION
**Pull request by ibuclaw**
This is neccesary for fixing gcd to be able to handle negative values (Issue 21834).

### CODE REVIEW

File: std/numeric.d [hunk 1]
```diff
@@ -2975,6 +2983,40 @@ if (isIntegral!T)
...
+            assert(gcd(T(9),   U(6))   == 3);
+            assert(gcd(T(3),   U(4))   == 1);
+            assert(gcd(T(32),  U(24))  == 8);
+            assert(gcd(T(5),   U(6))   == 1);
+            assert(gcd(T(54),  U(36))  == 18);
+        }
+    }
+}
+
+@safe unittest
```
Comment (berni44) [inline std/numeric.d:3016]: A leftover?

**Review by berni44 — APPROVED** [review]

File: std/numeric.d (hunk 1 above)
Comment (ibuclaw) [reply std/numeric.d]: Oops, yes. :-)

File: std/numeric.d [hunk 2]
```diff
@@ -2928,19 +2928,20 @@ an efficient algorithm such as $(HTTPS en.wikipedia.org/wiki/Euclidean_algorithm
...
 Returns:
     The greatest common divisor of the given arguments.
  */
-T gcd(T)(T a, T b)
-if (isIntegral!T)
+typeof(Unqual!(T).init % Unqual!(U).init) gcd(T, U)(T a, U b)
+if (isIntegral!T && isIntegral!U)
 {
-    static if (is(T == const) || is(T == immutable))
+    static if (is(T == const) || is(T == immutable) || is(U == const) || is(T == immutable))
```
Comment (berni44) [inline std/numeric.d:2942]:
```suggestion
    static if (is(T == const) || is(T == immutable) || is(U == const) || is(U == immutable))
```

Comment (ibuclaw) [reply std/numeric.d:2942]: Ah yes, I was meant to test mixing const/immutable as well.

Comment (ibuclaw) [reply std/numeric.d]: There's no need to actually have this, using `CommonType(Unqual!T, Unqual!U)` instead for holding the intermediaries.

**Comment by berni44** [comment]
Is this good to go now? I think, with my new write rights, I'm allowed to add the merge label now...

**Comment by ibuclaw** [comment]
@berni44 - sure, go for it.  I'll fix it up again in the next PR for negative integers.

**Pull request merged**